internal/controller/dispatcher.go

```diff
@@ -93,34 +95,68 @@ func (rd *ResponseDispatcher) Dispatch(ctx context.Context, m protocol.Message,
 }
 
 type WorkDispatcherFactory struct {
-	reader *kafka.Reader
+	readerConfig *queue.ConsumerConfig
 }
 
-func NewWorkDispatcherFactory(reader *kafka.Reader) *WorkDispatcherFactory {
+func NewWorkDispatcherFactory(cfg *queue.ConsumerConfig) *WorkDispatcherFactory {
 	return &WorkDispatcherFactory{
-		reader: reader,
+		readerConfig: cfg,
 	}
 }
 
 func (fact *WorkDispatcherFactory) NewDispatcher(account, nodeID string) *WorkDispatcher {
 	log.Println("Creating a new work dispatcher")
 	return &WorkDispatcher{
-		account: account,
-		nodeID:  nodeID,
-		reader:  fact.reader,
+		account:      account,
+		nodeID:       nodeID,
+		readerConfig: fact.readerConfig,
 	}
 }
 
 type WorkDispatcher struct {
-	account string
-	nodeID  string
-	reader  *kafka.Reader
+	account      string
+	nodeID       string
+	readerConfig *queue.ConsumerConfig
 }
 
 func (wd *WorkDispatcher) GetKey() string {
 	return fmt.Sprintf("%s:%s", wd.account, wd.nodeID)
 }
 
-func (wd *WorkDispatcher) Dispatch(ctx context.Context) {
-	fmt.Println("This is when the work dispatcher would consume a job from the jobs kafka topic")
+func (wd *WorkDispatcher) Dispatch(ctx context.Context, c chan<- Work) {
```

Comment:
Let's think about renaming this method to more closely match what its doing.  Maybe something along the lines of StartDispatchingMessages?  I'm open to ideas.  I think of dispatch is a singular event and this is more of a continuous process.  What do you think?

Reply:
Yep a name change would be useful here.